File: august/week1/detect_capital.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        if word == word.upper():
            return True
        elif word == word.lower():
            return True
        elif word[0].upper() == word[0] and word[1:] == word[1:].lower():
            return True
        else:
            return False
        

if __name__ == "__main__":
    sol = Solution()
    print(sol.detectCapitalUse("USA"))
    print(sol.detectCapitalUse("FlaG"))

File: august/week1/Detect_Capital.cpp
#include<iostream>
#include<string>

using namespace std;

class Solution {
public:
    bool detectCapitalUse(string word) {
        if (word[0] >= 'A' && word[0] <= 'Z') {
            bool check = true;
            for (int i = 1; i < word.length(); i++) {
                if (word[i] < 'A' || word[i] > 'Z')
                    check = false;
            }
            if (check == true)
                return true;
            check = true;
            for (int i = 1; i < word.length(); i++) {
                if (word[i] < 'a' || word[i] > 'z')
                    check = false;
            }
            return check;
        }
        else if (word[0] >= 'a' && word[0] <= 'z') {
            bool check = true;
            for (int i = 1; i < word.length(); i++) {
                if (word[i] < 'a' || word[i] > 'z')
                    check = false;
            }
            return check;
        }
        else
            return false;
    }
};

int main()
{
    string word = "USA";
    Solution sol;
    bool ans = sol.detectCapitalUse(word);
    printf(ans ? "true\n" : "false\n");
    return 0;
}